tests/testthat/test-check_data.R

```diff
@@ -5,3 +5,56 @@ test_that("informative error is thrown if technologies are missing", {
     expected_technologies = c("A", "B", "C")
   ), "Missing technologies: B, C.")
 })
+
+
+# check_level_availability ------------------------------------------------
+test_that("informative error is thrown if levels are missing", {
+  expect_error(check_level_availability(
+    data = tibble::tibble(
+      a = c("A1", "A2"),
+      b = c("B1", "B2")
+    ),
+    expected_levels_list = list(
+      a = c("A1"),
+      b = c("B1", "B2", "B3")
+    )
+  ), "Expected levels are B1, B2, B3")
+})
+
+
+# check_sector_tech_mapping -----------------------------------------------
+test_that("input is returned if mapping is correct", {
+  test_data <- tibble::tibble(
+    sector = c("Oil&Gas", "Oil&Gas"),
+    technology = c("Oil", "Gas")
+  )
+
+  test_data_checked <- check_sector_tech_mapping(
+    data = test_data,
+    sector_col = "sector"
+  )
+
+  expect_equal(test_data, test_data_checked)
+})
+
+test_that("error is thrown in case technology expected for ald_sector is missing.", {
```

Comment:
If a portfolio is only invested in a company that extracts Oil but not Gas, would this fail?

Reply:
nope, the check is only appleid to scenario and price data, so should not care about the portfolio

Reply:
fantastic